### ISSUE TITLE
Final Preparations & Evaluation Pipeline Validation

### ISSUE THREAD

**Issue by kutayeroglu** [kutayeroglu/ijepa#22]
Match evaluation performance presented in the paper for linear probe smallest 

Following is a quote from the original study:

> To evaluate the I-JEPA on ImageNet [60], we adapt the VISSL recipe to use average pooled repre-
sentations instead of the [cls] token. Following MAE [36], we use the LARS [72] optimizer with a batch-size of 16384, and train the linear probe for 50 epochs. We use a learning rate with a step-wise decay, dividing it by a factor of 10 every 15
epochs, and sweep three different reference learning rates [0.01, 0.05, 0.001], and two weight decay values [0.0005, 0.0].

**Comment by kutayeroglu** [comment]
I am currently using ViT-H patch_size:14x14 resolution:224x224 epochs:300
Original study reports an accuracy %79.3 for this architecture. 
Got **%69.22** on validation set with non-optimized training code w/%50 percent of the training set.

```
2025-11-11 21:06:36,007 [__main__:INFO] Loaded model successfully from: /users/kutay.eroglu/projects/ijepa/pretrained_models/IN1K-vit.h.14-300e.pth.tar
2025-11-11 21:06:43,153 [__main__:INFO] Created model with linear head
2025-11-11 21:06:45,576 [src.utils.linprobe_trainer:INFO] Starting linear probe training
2025-11-11 21:06:45,576 [src.utils.linprobe_trainer:INFO] Hyperparameters: lr=0.01, epochs=5
Epoch 1/5 [Train]: 100%|██████████| 125/125 [46:56<00:00, 22.53s/it, acc=63.29%, loss=0.0020]
Epoch 1/5 [Val]: 100%|██████████| 5/5 [01:55<00:00, 23.12s/it, acc=69.22%]
2025-11-11 21:55:38,103 [src.utils.linprobe_trainer:INFO] Epoch 1/5 | Train Acc: 63.29% | Val Acc: 69.22% | LR: 0.010000
2025-11-11 21:55:38,103 [src.utils.linprobe_trainer:INFO] New best model! (69.22% > 0.00%)
Epoch 2/5 [Train]: 100%|██████████| 125/125 [46:55<00:00, 22.52s/it, acc=70.39%, loss=0.0014]
Epoch 2/5 [Val]: 100%|██████████| 5/5 [01:55<00:00, 23.13s/it, acc=68.78%]
2025-11-11 22:44:29,318 [src.utils.linprobe_trainer:INFO] Epoch 2/5 | Train Acc: 70.39% | Val Acc: 68.78% | LR: 0.010000
Epoch 3/5 [Train]: 100%|██████████| 125/125 [46:55<00:00, 22.53s/it, acc=72.23%, loss=0.0013]
Epoch 3/5 [Val]: 100%|██████████| 5/5 [01:56<00:00, 23.21s/it, acc=69.02%]
2025-11-11 23:33:21,278 [src.utils.linprobe_trainer:INFO] Epoch 3/5 | Train Acc: 72.23% | Val Acc: 69.02% | LR: 0.010000
Epoch 4/5 [Train]: 100%|██████████| 125/125 [46:56<00:00, 22.53s/it, acc=73.39%, loss=0.0013]
Epoch 4/5 [Val]: 100%|██████████| 5/5 [01:55<00:00, 23.14s/it, acc=68.58%]
2025-11-12 00:22:13,396 [src.utils.linprobe_trainer:INFO] Epoch 4/5 | Train Acc: 73.39% | Val Acc: 68.58% | LR: 0.010000
Epoch 5/5 [Train]: 100%|██████████| 125/125 [46:56<00:00, 22.53s/it, acc=74.18%, loss=0.0012]
Epoch 5/5 [Val]: 100%|██████████| 5/5 [01:56<00:00, 23.22s/it, acc=68.38%]
2025-11-12 01:11:05,855 [src.utils.linprobe_trainer:INFO] Epoch 5/5 | Train Acc: 74.18% | Val Acc: 68.38% | LR: 0.010000
2025-11-12 01:11:05,855 [src.utils.linprobe_trainer:INFO] Training complete. Best validation accuracy: 69.22%
```
